lib/event_tracker.rb

```diff
@@ -0,0 +1,55 @@
+require './event_tracker_haystack.rb'
+
+divider
+
+
+puts "\n\n"
+puts "Requesting Koeln wiki from http://webmontag.de/location/koeln".green
+
+# requesting xhtmlbody file from keoln index.
+begin
+  webmontag_wiki_site = Nokogiri::HTML(open('http://webmontag.de/location/koeln/index?do=export_xhtmlbody'))
+  puts "Extracting website content".green
+rescue
+  puts "Cannot connect to website".red
+  abort
+end
+
+# Create archive object that has the whole li of links.
+# archive => Class: Nokogiri::XML::NodeSet
+#http://www.rubydoc.info/github/sparklemotion/nokogiri/Nokogiri/XML/NodeSet
+archive = webmontag_wiki_site.css('.archive li div')
+if (archive.empty?)
+  puts "\".archive\" selector is empty from the website!".red
+  abort
+end
+
+puts "\n\t #{archive.length} records found".green
+
+
+# This array contains each event from wiki website {name, href}.
+archive_clean = Array.new
```

Comment:
why are you using a new array here?  have you considered just modifying the original archive array?